step-strikers-app/step-strikers-app/Classes/RPGCharacter.swift

```diff
@@ -142,6 +156,151 @@ class Caster: RPGCharacter {
+    }
+    
+    func castInvisibility(caster: String, target: String, game: String) {
+        Firestore.firestore().collection("games").document(game).updateData([
+            "invisible": FieldValue.arrayUnion([target])])
+     
+        decreaseSpellPoints(amtDecrease: 10)
+        
+        // @Jalyn probably replace this with the logging strings you were talking about?
+        print("\(caster) cast invisibility on \(target)")
+    }
+    
+    func castMotivationalSpeech(caster: String, team: String) {
+        let teamRef = Firestore.firestore().collection("teams").document(team)
+        teamRef.getDocument { (document, error) in
+            if let document = document, document.exists {
+                let teammates = document.get("players") as! [String]
+                
+                for teammate in teammates {
+                    Firestore.firestore().collection("players").document(teammate).setData(["has_advantage": true], merge: true)
+                }
+            }
+        }
+        
+        decreaseSpellPoints(amtDecrease: 15)
+        
+        // @Jalyn probably replace this with the logging strings you were talking about?
+        print("\(caster) cast motivational speech on their team")
+    }
+    
+    func castAnimateDead(caster: String, target: String) {
+        Firestore.firestore().collection("players").document(target).setData(["health": 1], merge: true)
+        
+        decreaseSpellPoints(amtDecrease: 17)
+        
+        // @Jalyn probably replace this with the logging strings you were talking about?
+        print("\(caster) cast animate dead on \(target)")
+    }
+    
+    func heal(caster: String, target: String) {
+        let amtToHeal = rollDie(quant: 1, sides: 8)
+        
+        let targetRef = Firestore.firestore().collection("players").document(target)
+        targetRef.getDocument { (document, error) in
+            if let document = document, document.exists {
+                let currentHealth = document.get("health") as! Int
+                let maxHealth = document.get("max_health") as! Int
+                let newHealth = min(maxHealth, currentHealth + amtToHeal)
+                
+                Firestore.firestore().collection("players").document(target).setData(["health": newHealth], merge: true)
+            }
+        }
+        
+        decreaseSpellPoints(amtDecrease: 5)
+        
+        // @Jalyn probably replace this with the logging strings you were talking about?
+        print("\(caster) cast heal on \(target)")
+    }
+    
+    func sleep(caster: String, target: String) {
+        Firestore.firestore().collection("players").document(target).setData(["is_asleep": true], merge: true)
+        
+        decreaseSpellPoints(amtDecrease: 12)
+        
+        // @Jalyn probably replace this with the logging strings you were talking about?
+        print("\(caster) cast sleep on \(target)")
+    }
+
 }
     
 class Fighter: RPGCharacter {
```

Comment:
Due to the size of the RPGCharacter class, please add a Subclasses.swift that holds the Fighter, Bard, Rogue, and Wizard classes